About Page: remove submenu and add link to page in footer

jeherve opened the pull request:
Fixes #12315

#### Changes proposed in this Pull Request:

You can now access the About Page from 4 different locations:

- The Automattic link at the bottom of the Jetpack dashboard pages.
- The About link at the bottom of the Jetpack dashboard pages.
- The Automattic logo on any Jetpack non-react admin pages (e.g. Settings > Sharing)
- The about link at the bottom of any Jetpack non-react admin pages (e.g. Settings > Sharing)

#### Testing instructions:

* Go to any of the pages I mentioned above?
* Do you see the links?
* Can you click on them, does the page open in the same tab?

**Note**: 
- If your site is connected to wpcom or in dev mode, the About link will lead to the in-plugin About page.
- If you are not connected yet, the About link will lead to Jetpack.com.

#### Proposed changelog entry for your changes:

* None

jeherve committed:
About Page: remove submenu and add link to page in footer

Fixes #12315

You can now access the About Page from 4 different locations:

- The Automattic link at the bottom of the Jetpack dashboard pages.
- The About link at the bottom of the Jetpack dashboard pages.
- The Automattic logo on any Jetpack non-react admin pages (e.g. Settings > Sharing)
- The about link at the bottom of any Jetpack non-react admin pages (e.g. Settings > Sharing)

eliorivero commented on _inc/lib/admin-pages/class.jetpack-admin-page.php:
```diff
@@ -315,11 +315,14 @@ public static function wrap_ui( $callback, $args = array() ) {
 						<a href="https://jetpack.com" target="_blank" rel="noopener noreferrer" class="jp-footer__link" title="<?php esc_html_e( 'Jetpack version', 'jetpack' ); ?>">Jetpack <?php echo JETPACK__VERSION; ?></a>
 					</li>
+					<li class="jp-footer__link-item">
+						<a href="<?php echo esc_url( admin_url() . 'admin.php?page=jetpack_about' ); ?>" title="<?php esc_attr__( 'About Jetpack', 'jetpack' ); ?>" class="jp-footer__link"><?php echo esc_html__( 'About', 'jetpack' ); ?></a>
```
```suggestion
						<a href="<?php echo esc_url( admin_url( 'admin.php?page=jetpack_about' ) ); ?>" title="<?php esc_attr__( 'About Jetpack', 'jetpack' ); ?>" class="jp-footer__link"><?php echo esc_html__( 'About', 'jetpack' ); ?></a>
```

The path should be inside `admin_url`. See https://codex.wordpress.org/Function_Reference/admin_url#Examples

jeherve replied:
Good call. Done in 698087716a7481f72a21080e2f4e6cff4c3faba3

eliorivero reviewed (changes requested):
Minor change requested, added code suggestion

eliorivero commented:
About page is no longer in the menu ✅ 

<img width="250" alt="Captura de Pantalla 2019-05-09 a la(s) 15 42 48" src="https://user-images.githubusercontent.com/1041600/57478435-440bb680-7271-11e9-846c-e567359e56e8.png">

It's in the footer and properly linked ✅ 

<img width="348" alt="Captura de Pantalla 2019-05-09 a la(s) 15 43 05" src="https://user-images.githubusercontent.com/1041600/57478436-440bb680-7271-11e9-834b-6e099481a8f2.png">

Goes to blank page ❗️ 

<img width="551" alt="Captura de Pantalla 2019-05-09 a la(s) 15 43 26" src="https://user-images.githubusercontent.com/1041600/57478437-44a44d00-7271-11e9-9e0e-f6bafc338f4d.png">

jeherve committed:
Move path inside admin_url

Props @eliorivero
https://github.com/Automattic/jetpack/pull/12323#discussion_r282607565

jeherve committed:
Only display link to About page when it is accessible.

- If your site is connected to wpcom or in dev mode, the About link will lead to the in-plugin about page.
- If you are not connected yet, the About link will lead to Jetpack.com.

See https://github.com/Automattic/jetpack/pull/12323#issuecomment-491021123

jeherve commented:
Thanks for the feedback. This should be good for another review!

eliorivero commented:
🐑
